tests/blackbox/stratis_cert.py

```diff
@@ -0,0 +1,347 @@
+        StratisCli.fs_create(pool_name, fs_name)
+        StratisCli.fs_create(pool_name, fs_too)
+        StratisCli.fs_destroy(pool_name, fs_name)
+
+        fs = StratisCli.fs_list()
+        self.assertEqual(1, len(fs))
+        self.assertFalse(fs_n in fs)
+        self.assertTrue(fs_too in fs)
+
+    def test_fs_snap_shot(self):
+        """
+        Test creating a FS snap shot.
+        :return: None
+        """
+        pool_name = p_n()
+        fs_name = fs_n()
+        fs_ss = fs_n()
+        StratisCli.pool_create(pool_name, [DISKS[0]])
+        StratisCli.fs_create(pool_name, fs_name)
+        StratisCli.fs_ss_create(pool_name, fs_name, fs_ss)
+        fs = StratisCli.fs_list()
+        self.assertTrue(fs_ss in fs)
+
+    def test_block_dev_list(self):
+        """
+        Test block device listing
+        :return:
+        """
+        pool_name = p_n()
+        StratisCli.pool_create(pool_name, [DISKS[0]])
+
+        block_devs = StratisCli.blockdev_list()
+        self.assertTrue(DISKS[0] in block_devs)
+        self.assertEqual(1, len(block_devs))
+        self.assertEqual(pool_name, block_devs[DISKS[0]]["POOL_NAME"])
+
+    @unittest.expectedFailure
+    def test_no_args(self):
+        """
+        Ref. https://github.com/stratis-storage/stratis-cli/issues/248
+        :return: None
+        """
+        exec_command([STRATIS_CLI], 2)
+        exec_command([STRATIS_CLI, "daemon"], 2)
+
+    def test_fs_rename(self):
+        """
+        Test renaming a FS
+        :return: None
+        """
+        pool_name = p_n()
+        fs_name = fs_n()
+        fs_new_name = fs_n()
+        StratisCli.pool_create(pool_name, [DISKS[0]])
+        StratisCli.fs_create(pool_name, fs_name)
+        StratisCli.fs_rename(pool_name, fs_name, fs_new_name)
+
+        fs = StratisCli.fs_list()
+        self.assertTrue(fs_new_name in fs.keys())
+        self.assertFalse(fs_name in fs.keys())
+        self.assertEqual(1, len(fs))
+
+    @unittest.expectedFailure
+    def test_signal_interruption(self):
+        """
+        Send a signal in the middle of a command to ensure that we don't get
+        a stack trace, ref. https://bugzilla.redhat.com/show_bug.cgi?id=1686652
+        :return:
+        """
+        process = Popen(
+            [STRATIS_CLI, "pool", "create",
+             p_n(), DISKS[0]],
+            stdout=PIPE,
+            stderr=PIPE,
+            close_fds=True,
+            env=os.environ)
+        time.sleep(0.05)
+        process.send_signal(2)
+        result = process.communicate()
+        stdout_text = ""
+        stderr_text = ""
+        if result[0]:
+            stdout_text = bytes(result[0]).decode("utf-8")
+        if result[1]:
+            stderr_text = bytes(result[1]).decode("utf-8")
+
+        self.assertTrue("Traceback" not in stdout_text)
+        self.assertTrue("Traceback" not in stderr_text)
+        self.assertNotEqual(process.returncode, 0)
+
+    @unittest.expectedFailure
```

Comment:
Please change this test. It hasn't been agreed that putting the name in the error is the appropriate thing, see comment: https://bugzilla.redhat.com/show_bug.cgi?id=1686852#c1.

Reply:
Correct no determination has been made, thus we can leave it as is because we are expecting it to fail.  When we expect it to pass we can then make the final changes to what it should be checking for.

Reply:
I think it is better that the test test for the most important thing, which is that if the pool is created with disks that have already been claimed by another pool, then the command will return an exit code of 1. Because this test is marked as an expected failure, if that becomes not true, the test will still succeed. The test is most valuable if the ```expectedFailure``` and the following lines are stripped out. 
```
        # Users want the pool name in the error output
        self.assertTrue(pool_name in stdout)
```
and is left as a normal running test.

Checks for what the error messages actually contains can be added later to refine the test, if that seems like the correct thing to do.

Reply:
All of the tests that are decorated with `@unittest.expectedFailure` are for regression tests for existing bugs.  The sole purpose of this test is to ensure that the pool name is in the output.  In my opinion the end user is correct that the pool name should be included in the error message.  I believe this is the correct thing to do because everything in the CLI is name based.  Users never specify anything by UUID, never see anything by UUID, thus it makes sense that the error message would specify the pool by name.  Having both the UUID and name in the output would be fine in my opinion too.